### PR TITLE
Docker workflow

### PR DESCRIPTION
**Pull request by franchuterivera**
This push creates a workflow so that when a push to a branch called:
      - master
      - development
      - test_docker

happens, a docker image is created and pushed to the github registry as a package.

Also, added documentation for this. Some permission inside of the .github file had to be changed for that.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
```
Comment (mfeurer) [inline .github/ISSUE_TEMPLATE/bug_report.md:1]: I don't think this file should be part of this PR, right?

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Push to GitHub Packages
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: docker.pkg.github.com
+          repository: automl/auto-sklearn/auto-sklearn
+          tag_with_ref: true
```
Comment (mfeurer) [inline .github/workflows/docker-publish.yml:25]: This looks great for building. Would it also be possible to run the unit tests in this repo after building the container so we are certain that Auto-sklearn works inside the container?

**Comment by franchuterivera** [comment]
Hi Matthias, I think you have to run it to see the new workflow.

In my testing, it looks like this: https://github.com/franchuterivera/auto-sklearn/pull/4/checks?check_run_id=879715992

File: Dockerfile [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
+
+# Copy the checkout autosklearn version for installation
+ADD . /auto-sklearn/
+
+# install linux packages
+RUN apt-get update
+
+# Set the locale
+RUN apt-get -y install locales
+RUN touch /usr/share/locale/locale.alias
```
Comment (mfeurer) [inline Dockerfile:13]: Could you briefly explain what these lines are needed for?

File: Dockerfile [hunk 4]
```diff
@@ -1,19 +1,34 @@
...
-  | xargs -n 1 -L 1 pip3 install
+RUN pip3 install pytest==4.6.* pep8 codecov pytest-cov flake8 flaky openml
+RUN curl https://raw.githubusercontent.com/automl/auto-sklearn/master/requirements.txt | xargs -n 1 -L 1 pip3 install
+RUN pip3 install jupyter
 
 # Install
-RUN pip3 install \
-  auto-sklearn \
-  jupyter
+RUN pip3 install -e /auto-sklearn/
```
Comment (mfeurer) [inline Dockerfile:34]: Could you briefly explain what the `/` is doing here?

File: Dockerfile (hunk 3 above)
Comment (franchuterivera) [reply Dockerfile:13]: These lines are needed for the UTF encoding error during lazy import installation. Please let me know if you are aware of a better WA to overcome UnicodeDecodeErrors while pip installing this module

File: Dockerfile (hunk 4 above)
Comment (franchuterivera) [reply Dockerfile:34]: Oh, so I wanted to make sure we use the latest image available.

Before in the docker file:

1. We did a change in the development repo. We install autosklearn from pip. You don't get the change in the development image.

Workflow now:
1- You checkout the development branch (or any branch in question) in /auto-sklearn of the container
2- You install autosklearn from the checkout version and successfully get the change you implemented.

Comment (mfeurer) [reply Dockerfile:34]: Got it, thanks.

**Pull request merged**